### PR TITLE
dev-friendships-v2 Pull Request

### PR DESCRIPTION
**Pull request by FSolM**
Sixth milestone pull Request 👤

### CODE REVIEW

**Comment by codingAngarita** [comment]
# Requested changes:

## Linting
>      Linting is still not present in the project

We have decided to use `rubocop` for linting, we added a file `.rubocop.yml` that contains all the configs used.

## Tests
>     Tests already have some failed examples.

I wasn't able to replicate this behavior, currently, this is the output I get when I run `$ rails spec`:

`Finished in 1.31 seconds (files took 4.88 seconds to load)
82 examples, 0 failures`

## Issues
>     There is an error when clicking the Friendship link in the navbar, see #12 

Both issues #12 and #13 have been addressed.

## Friends index
>      Link to check all the users isn't present so I can't do a friend request.
We don't understand this requirement, are we supposed to show **ALL** the application registered users?

**Comment by codingAngarita** [comment]
# Requested changes:
Hi @mcrd25!
Thank you for your positive and helpful code reviews :+1:

## Stickler
We added the `.stickler.yml` file with the config needed and @FSolM enabled stickler on the repo, however, It's not displaying correctly, this is the output I get when I run `rubocop` locally:

image

## Specs
The test cases were still passing on my local env, this is the output I get when I run `rails spec`:

image

# Additional suggestions
We addressed the additional suggestions, both really cool and useful! 

* Specifying the date required is the DOB:
image

* Making the friendships method be `index` instead of `show`:
image

**Comment by azdrenymeri** [comment]
## Hello @FSolM :wave: 
#### You really did a great job on this project :1st_place_medal: 
* There is just one thing left to check 
- [ ] on your posts feeds you should include your posts and your friends posts only:
- Janed, Azdren and John are not friends with each other and they still can see each others posts.
Screenshot

### Fix this small issue and add it for a re-review

**Comment by FSolM** [comment]
The idea of the feed is more akin to a general post hub rather than just friends. We mash in together various ideas of a social network instead of just grabbing from one source. Think of our feed as a Reddit feed preferably than a Facebook one

File: app/controllers/user/registrations_controller.rb [hunk 1]
```diff
@@ -1,65 +1,66 @@
...
+    def configure_sign_up_params
+      devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
+    end
 
-  # If you have extra params to permit, append them to the sanitizer.
-  # def configure_account_update_params
-  #   devise_parameter_sanitizer.permit(:account_update, keys: [:attribute])
-  # end
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_account_update_params
```
Comment (mcrd25) [inline app/controllers/user/registrations_controller.rb]: Hey,
It’s better to delete commented code. If you need to review a previous version of your code, use git history.
Please remove.

File: app/controllers/user/registrations_controller.rb [hunk 2]
```diff
@@ -1,65 +1,66 @@
...
+    #   devise_parameter_sanitizer.permit(:account_update, keys: [:attribute])
+    # end
 
-  # The path used after sign up.
-  # def after_sign_up_path_for(resource)
-  #   super(resource)
-  # end
+    # The path used after sign up.
+    # def after_sign_up_path_for(resource)
```
Comment (mcrd25) [inline app/controllers/user/registrations_controller.rb]: Hey,
It’s better to delete commented code. If you need to review a previous version of your code, use git history.
Please remove.

File: app/controllers/user/registrations_controller.rb [hunk 3]
```diff
@@ -1,65 +1,66 @@
...
+    #   super(resource)
+    # end
 
-  # The path used after sign up for inactive accounts.
-  # def after_inactive_sign_up_path_for(resource)
-  #   super(resource)
-  # end
+    # The path used after sign up for inactive accounts.
+    # def after_inactive_sign_up_path_for(resource)
```
Comment (mcrd25) [inline app/controllers/user/registrations_controller.rb]: Hey,
It’s better to delete commented code. If you need to review a previous version of your code, use git history.
Please remove.

File: app/controllers/user/sessions_controller.rb [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+      @recent_posts = Post.last(2).reverse
+      super
+    end
 
-  # POST /resource/sign_in
-  # def create
-  #   super
-  # end
+    # POST /resource/sign_in
+    # def create
```
Comment (mcrd25) [inline app/controllers/user/sessions_controller.rb]: Hey,
It’s better to delete commented code. If you need to review a previous version of your code, use git history.
Please remove.
Please do the same for all other commented out code before next review.

**Comment by codingAngarita** [comment]
## Thanks @mcrd25 
The commented out code has been removed :tada: and now we're working with heroku do deploy the live version.

**Comment by FSolM** [comment]
I'm crying rn .-.

**Comment by mcrd25** [comment]
WOW

**Pull request merged**